Userland/Libraries/LibWeb/XHR/XMLHttpRequest.cpp

```diff
@@ -241,11 +238,16 @@ WebIDL::ExceptionOr<JS::Value> XMLHttpRequest::response()
             return JS::js_null();
 
         // 4. Set this’s response object to jsonObject.
-        m_response_object = JS::NonnullGCPtr<JS::Object> { json_object_result.release_value().as_object() };
+        if (json_object_result.value().is_object())
```

Comment:
maybe this could be considered a spec issue? :thonk:

Reply:
I think this is just an artifact of how we represent the "response object" in XMLHttpRequest.
`Variant<JS::NonnullGCPtr<JS::Object>, Failure, Empty>` is a little odd, but I suppose it makes sense if we need to distinguish between Failure and Empty. :)